pkg/kubelet/kuberuntime/kuberuntime_sandbox.go

```diff
@@ -143,6 +144,9 @@ func (m *kubeGenericRuntimeManager) generatePodSandboxConfig(pod *v1.Pod, attemp
 }
 
 // generatePodSandboxLinuxConfig generates LinuxPodSandboxConfig from v1.Pod.
+// We've to call PodSandboxLinuxConfig always irrespective of the underlying OS as securityContext is not part of
```

Comment:
Good point - it seems there is no point in calling `generatePodSandboxLinuxConfig` for Windows and therefore the `runtime.GOOS != "windows"` checks below can be moved [up](https://github.com/kubernetes/kubernetes/blob/428b500c5a93daded135f347428724334163859a/pkg/kubelet/kuberuntime/kuberuntime_sandbox.go#L140) to the caller? Note that in case of Windows, the relevant security context details (covering GMSA credspec details and runAsUsername) is passed as part of the [container security context]( https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto#L789 ) (rather than pod/sandbox level context)

Reply:
Got it. I have filed https://github.com/kubernetes/kubernetes/issues/93580 to keep track of the followup around this.

Reply:
Thank you,